src/util/tls.ts

```diff
@@ -12,12 +12,14 @@ export type HttpsOptions = {
     key: string;
     cert: string;
     keyLength?: number;
+    domain?: string;
```

Comment:
For clarity, I think this should be `defaultDomain`, and it would be useful if you could add a little jsdoc documentation here, just explaining what it does: this is the domain name that will be used in the certificate for incoming TLS connections which don't use SNI to request a specific domain.

Same in the case below (or bonus points if you can split keyLength & domain out into a separate object neatly somewhere).